### DISCUSSION TITLE
Custom easing function for animation

### DISCUSSION THREAD

**Discussion by hrishikesh-k** [unocss/unocss#2305] [Q&A] [answered]
Hello!

I'm trying to find a way to add a custom animation-timing-function value to the Uno Config, but so far I have been unsuccessful. My latest attempt is [here](https://uno.antfu.me/play/?html=DwZwbg5gUABDCuBaAhgOwJYFtkBcCmAvAEQAm8ATrugPaqICMALAAzMzr6U413qoBmfDnhipkmPIgA2fSfRjdMfCADE6M1HKKwYYdHgDuAIWoAPYmzb0ATAA4YN20QB8O4AAdcACx1wkgqSlifGQpRBZmbTg4EmIAWQBWWwA6AE5GABoAZkZkgHYEgDUUxmspfKzECq8U23K8rIyKsGsANmSsrLCKpoavRFruhqqGgC0iAHpXYAnwCGcgA&config=JYWwDg9gTgLgBAbwCYFMBmwB2KDCFMYDmANHGFCgM4owCCMMUwARgK4zBoCep5VNAVUwQAvnDRQIIOAHJWwgMaVKMgFAoAHpFhxUaAIasANvD1Zc%2BIgAoEquGQrUYlAFxwA2nfsP%2BdBkzYObhsvbzhhTAA1fSNWFCR6RhZ2FDcDI2piUO8%2BDA03OQBaGSyw%2B1zgDXiAeUwjLjdGONKyyiSFGEaoOOyRAEoW8sdBYRCy3X0oAGsChSN9ZRLs%2BwA3SeB9ZiMUAAUKPILWYt6%2BrwBdFpgACxQQVMRshQgjaFcH8ZgUGLdbcfsZAAsAAYgTICgBiIEAVmYSGY%2BjU4xEvUGcH0mFA%2Bg4%2BB%2ByzgHBAWEIADFMG9fn8ZEZzIUAIxg2QKVjMYAKQrMFAAL2AKCgViBADoAGwADlIhUFQtIgoAzAAmUi0gVQvqIsoiVHI%2BzI-pAA&css=AIawpgngZgTghgWzAZwAQBsCWA7MBaARlQG8AoVVABgFITyLUB7ABzgGNMAXCALioG56FTvGzIojGAj4i4Y9HE5gAmgAo8lAHQAmAKwwwCAJSCKAX3oA2GnQZNWHbnwKmGssRKkzRyBUrWUJvQWFASUNmR27uKS0qjRfiqq2pr6hkHmpGZAA&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwF8g). I took some inspiration from [this discussion](https://github.com/unocss/unocss/discussions/2012) that helped me find `animation.timingFn`, but not sure how to specify only that value. I don't want to specify the entire animation in there.

I also tried using `ease-` instead of `timingFn-` in my HTML. The default `ease` values like `ease-in` work fine. But I need the custom function in there.

Also, I need to use the same timingFn for multiple animations. So, I'd prefer not having to hardcode that with the animation name, instead just use timingFn independently.

Is it possible?

**Comment by zyyv** [comment, marked as answer]
Sorry, animate timing function use `theme.ease` field.
[Playground](https://uno.antfu.me/play/?html=DwZwbg5gUABDCuBaAhgOwJYFtkBcCmAvAEQAm8ATrugPaqICMALAAzMzr6U413qoBmfDnhipkmPIgA2fSfRh5kISTNR4isGGHR4A7gCFqAD2Js29AEwAOGJatEAfJuAAHXAAtNcJIKlTi%2BMhSiCzMGnBwJMQAsgCsVgB0AJyMADQAzIwJAOyxAGqJjBZSOemIpe6JViXZ6amlYBYAbAnp6cGl9bXuiFUdteW1AFpEAPROwKPgEA5AA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmANHGFKgM6owCCMMUwARgK4zDoCeZF1tAKpYIcAL5x0UCCDgByNiIDGVKnIBQqAB6RYKDAEM2AG3hpMOfIWBEAFAnVxylGjCoAuOAG1HT5wPpGZnZOHntfPzgRLAA1A2M2VGQGJlYOVE90eJoSCL9%2BTC1PBQBaOVzIpwLgLSSAeSxjbk8mRIrKqlSlGBaoNryxAEp2qpchEXDKlAMoAGtipWMDVXK8pwA3GeADFmNUAAVKQuK2MpHxYd8AXXaYAAtUEAzEPNRl7CJPBynjbGe5JRsFjAJQlFioABewFQUFsAAYAHQANgAHGQSoikWREQBmABMZAAjAiAKyDVaRMTnJQQYzQDwvKYwN7GL5rPwAFjhcOKAGI4SSWMgWAYKZUqQN2hKhuogA&css=AIawpgngZgTghgWzAZwAQBsCWA7MBaARlQG8AoVVABgFITyLUB7ABzgGNMAXCALioG56FTvGzIojGAj4i4Y9HE5gAmgAo8lAHQAmAKwwwCAJSCKAX3oA2GnQZNWHbnwKmGssRKkzRyBUrWUJvQWFASUNmR27uKS0qjRfiqq2pr6hkHmpGZAA&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwF8g)

https://github.com/unocss/unocss/blob/4959845937841a21e324a4beb2fb96a84f97a751/packages/preset-wind/src/rules/animation.ts#L38

**Reply by hrishikesh-k** [reply]
That was it, thank you!

**Reply by Alberto0738gh** [reply]
How do
